src/OrchardCore/OrchardCore.Media.Core/DefaultMediaFileStoreCacheFileProvider.cs

```diff
@@ -66,7 +66,12 @@ public async Task SetCacheAsync(Stream stream, IFileStoreEntry fileStoreEntry, C
                 }
                 using (var fileStream = File.Create(cachePath))
                 {
-                    await stream.CopyToAsync(fileStream, StreamCopyBufferSize, cancellationToken);
+                    await stream.CopyToAsync(fileStream, StreamCopyBufferSize);
+
```

Comment:
can you add a `await stream.FlushAsync()` here as well?

Because we realised the dispose would not be the async